### PR TITLE
fix issue 201 can use hyphen in name

### PR DESCRIPTION
**Pull request by askazakov**
fix https://github.com/ardalis/CleanArchitecture/issues/201

see https://github.com/dotnet/templating/issues/1168

now I can
```
mkdir try-temp.1 && cd "$_"
dotnet new clean-arch
```

### CODE REVIEW

**Comment by askazakov** [comment]
a lot of changes but there is only renaming actually

**Comment by ardalis** [comment]
Looking to merge this; can you resolve conflicts when you get a chance? Sorry for the delay in getting to it

**Comment by askazakov** [comment]
I merge main but encountered with https://github.com/ardalis/CleanArchitecture/pull/308

**Comment by askazakov** [comment]
@ardalis should I bump version to 6.0.10?

File: .github/workflows/dotnetcore.yml [hunk 1]
```diff
@@ -20,4 +20,4 @@ jobs:
     - name: Build with dotnet
       run: dotnet build --configuration Release
     - name: Test with dotnet
-      run: dotnet test ./Clean.Architecture.sln --configuration Release
+      run: dotnet test ./Clean.Architecture.1.sln --configuration Release
```
Comment (ardalis) [inline .github/workflows/dotnetcore.yml:23]: What are the `.1` additions for?

Comment (askazakov) [reply .github/workflows/dotnetcore.yml:23]: .sln file name should match with `"sourceName"` from `template.json`. `.1` is one option to respect special symbol (like `-`) to correctly instantiate solution from template. Another option is to use `_Clean.Architecture_` everywhere

**Pull request closed without merging**